src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs

```diff
@@ -28,19 +29,43 @@ internal static AcquireTokenForClientParameterBuilder WithAtPop(
 
             builder.WithProofOfPosessionKeyId(popPublicKey);
             builder.OnBeforeTokenRequest((data) =>
-             {
-                 string? signedAssertion = GetSignedClientAssertion(
-                     clientCertificate,
-                     data.RequestUri.AbsoluteUri,
-                     jwkClaim,
-                     clientId, 
-                     sendX5C);
+            {
+                string? signedAssertion = GetSignedClientAssertion(
+                    clientCertificate,
+                    data.RequestUri.AbsoluteUri,
+                    jwkClaim,
+                    clientId,
+                    sendX5C);
+
+                data.BodyParameters.Remove("client_assertion");
+                data.BodyParameters.Add("request", signedAssertion);
+
+                return Task.CompletedTask;
+            });
+
+            return builder;
+        }
+
+        // Configure with Custom Signed Assertion
+        internal static AcquireTokenForClientParameterBuilder WithAtPop(
+            this AcquireTokenForClientParameterBuilder builder,
+            CredentialDescription credentialDescription,
+            string popPublicKey,
+            string jwkClaim)
+        {
+            _ = Throws.IfNull(popPublicKey);
+            _ = Throws.IfNull(jwkClaim);
 
-                 data.BodyParameters.Remove("client_assertion");
-                 data.BodyParameters.Add("request", signedAssertion);
+            builder.WithProofOfPosessionKeyId(popPublicKey);
+            builder.OnBeforeTokenRequest((data) =>
+            {
+                string? signedAssertion = credentialDescription.CachedValue as string;
```

Comment:
I don't understand that. Why not let MSAL use the client assertion that was provided with the delegate at the construction of the confidential client application instance. I thought that the whole point of the new MS Auth 1.0 at pop was to be able to allow for FIC etc ?

Are lines 62 and 63 really needed? and also do we need to pass-in the `credentialDescription` ?

Reply:
Removed custom signed assertion configuration.
It will now be set up via delegate during CCA initialization.